projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid-navigation.service.ts

```diff
@@ -174,6 +175,23 @@ export class IgxPivotGridNavigationService extends IgxGridNavigationService {
                 layout: null
             }
 
+            if (ctrl) {
+                const dimIndex = this.activeNode.column;
+                const flatRowDimensions = PivotUtil.flatten(this.grid.visibleRowDimensions)
```

Comment:
There's no need to use `flatten` in this case. The `visibleRowDimensions` contains a flat structure of all shown row dimensions.